code/modules/tgui/external.dm

```diff
@@ -213,5 +213,9 @@
 		payload = json_decode(href_list["payload"])
 	// Pass message to window
 	if(window)
+		//this proc is called from /client/Topic() which is called at the very end of a tick like a verb, so try to queue this high cost proc if
+		//the sever is overloaded.
+		if(TRY_QUEUE_VERB(window, /datum/tgui_window.proc/on_message, type, payload, href_list))
+			return TRUE
```

Comment:
Might potentially worsen the current state of tgui, because window initialization is time sensitive. It's better to move TRY_QUEUE_VERB towards ui_act part, since that is a less critical path and can be the most cpu intensive.

Reply:
https://github.com/tgstation/tgstation/blob/master/code/modules/tgui/tgui.dm#L311